### DISCUSSION TITLE
[@types/node] AbortController.abort() is missing optional reason parameter

### DISCUSSION THREAD

**Discussion by laurenzhonauer** [DefinitelyTyped/DefinitelyTyped#58273] [Issues with a @types package]
## Description

In the definition of the [AbortController on MDN](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort) it is stated, that it is possible to pass a parameter to `AbortController.abort()` that is thrown as Error if the AbortController is used in a fetch call.

## Expected

Like done in the [@types/mongodb] implementation ([see here](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44549)), the optional reason parameter is also needed in the [@types/node] implementation.

## Actual

`AbortController.abort()` exists, but does not expect a parameter.

[a comment by an automated account is not shown]

**Comment by laurenzhonauer** [comment]
Im not sure if I understood the implications right, but I also opened an issue in [node-fetch](https://github.com/node-fetch/node-fetch/issues/1462) regarding the same issue.

I hope thats correct, if not please help me connect the right dots 😄

**Comment by peterblazejewicz** [comment]
I think this is due to a fact, both Node and `dom` types are kept in sync. Thanks to this same api can be used without causing an errors (in theory). This requires using declaration merging and keeping in sync Node types with standard dom lib. For now it seems it does not take any param (I know MDN takes any object as the token):
https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L1900-L1909
This can be proper question for TS itself

**Reply by laurenzhonauer** [reply]
Thanks for the pointer. I opened the same issue [in the typescript project](https://github.com/microsoft/TypeScript/issues/47505)

**Reply by tonivj5** [reply]
@laurenzhonauer your opened issue was resolved. @peterblazejewicz I think the issue is now with @types/node (ref https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/64570)
